### PR TITLE
Update Developer Guide for Edit Commands and Architecture Diagrams

### LINKED ISSUE #47: Update Developer Guide
On Architecture Diagrams and EditCommands implementations.

### PR DESCRIPTION
**Pull request by aaronsms**
So for the architecture diagrams, nothing has changed actually, so I will use the same diagrams as before. For the edit commands implementation, I might add a sequence diagram, but that would be very similar to other commands like Add and Delete. I would feel that an object diagram, to explain the dependencies between the models would be more relevant to my content, but that might be covered under the model designs?

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -329,7 +413,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Glossary
 
-* **Mainstream OS**: Windows, Linux, Unix, OS-X
```
Comment (daniellau88) [inline docs/DeveloperGuide.md]: I think the first asterick is for bullet point

**Comment by daniellau88** [comment]
> So for the architecture diagrams, nothing has changed actually, so I will use the same diagrams as before. For the edit commands implementation, I might add a sequence diagram, but that would be very similar to other commands like Add and Delete. I would feel that an object diagram, to explain the dependencies between the models would be more relevant to my content, but that might be covered under the model designs?

But you would have to come up with something that you have worked with before, so I think would be best to add the sequence diagram

**Comment by aaronsms** [comment]
[Architecture Diagram](https://github.com/aaronsms/tp/blob/edit-commands-dg-update/docs/images/Architecture.png)
[Architecture Sequence Diagram](https://github.com/aaronsms/tp/blob/edit-commands-dg-update/docs/images/ArchitectureSequence.png)
[EditCommand Sequence Diagram](https://github.com/aaronsms/tp/blob/edit-commands-dg-update/docs/images/EditCommandSequenceDiagram.png)

**Comment by daniellau88** [comment]
For the `EditCommandSequenceDiagram`, do include the command string

**Comment by aaronsms** [comment]
> For the `EditCommandSequenceDiagram`, do include the command string

Yeap done!

**Review by weixue123 — APPROVED** [review]

**Pull request merged**